src/server/handlers/consents/{ID}.ts

```diff
@@ -0,0 +1,93 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files
+ are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Ahan Gupta <ahangupta@google.com>
+
+ --------------
+ ******/
+import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
+import { Consent } from '../../../model/consent'
+import { Logger } from '@mojaloop/central-services-logger'
+import { retrieveValidConsent, updateConsentCredential, putConsents, ConsentCredential, checkCredentialStatus } from '../../../domain/consents/consents'
+import { IncorrectChallengeError, IncorrectStatusError } from '../../../domain/errors'
+import { verifySignature } from '../../../lib/challenge'
+import { NotFoundError } from '../../../model/errors'
+import { Enum } from '@mojaloop/central-services-shared'
+
+export async function retrieveUpdateAndPutConsent (id: string, challenge: string,
+  credentialStatus: string, signature: string,
+  publicKey: string, requestCredentialId: string,
+  request: Request): Promise<void> {
+  let consent: Consent
+  try {
+    consent = await retrieveValidConsent(id, challenge)
+    /* Checks if incoming credential status is of the correct form */
+    await checkCredentialStatus(credentialStatus, id)
```

Comment:
do we need a sep function for a 2 line operation?

Reply:
yes, if it:
- improves readability 
- hides implementation details 
- changes level of abstract